IS0402Test.py

```diff
@@ -1803,19 +1801,28 @@ def test_29(self):
         """Query API supports websocket subscription request"""
         test = Test("Query API supports request of a websocket subscription")
 
-        api = self.apis[REG_API_KEY]
+        api = self.apis[QUERY_API_KEY]
 
-        if self.is04_reg_utils.compare_api_version(api["version"], "v2.0") < 0:
+        if self.is04_query_utils.compare_api_version(api["version"], "v2.0") < 0:
             sub_json = deepcopy(self.subscription_data)
-            if self.is04_reg_utils.compare_api_version(api["version"], "v1.2") < 0:
+            sub_json["secure"] = ENABLE_HTTPS
+            if self.is04_query_utils.compare_api_version(api["version"], "v1.2") < 0:
                 sub_json = self.downgrade_resource("subscription", sub_json, self.apis[REG_API_KEY]["version"])
 
             valid, r = self.do_request("POST", "{}subscriptions".format(self.query_url), data=sub_json)
             if not valid:
                 return test.FAIL("Query API did not respond as expected")
             elif r.status_code == 200 or r.status_code == 201:
+                # Check protocol
+                response_json = r.json()
+                if not response_json["ws_href"].startswith(self.ws_protocol + "://"):
+                    return test.FAIL("WebSocket URLs must begin {}://".format(self.ws_protocol))
+                if self.is04_query_utils.compare_api_version(api["version"], "v1.1") >= 0:
```

Comment:
Picky, but I'd like to see the 'secure' check and the 'ws_href' check reordered, since I would treat the 'secure' flag as the fundamental (derived from the request), and the 'ws_href' protocol to be consistent with it.